### ISSUE TITLE
[Bug]: ut failed TestIterLocks

### ISSUE THREAD

**Issue by xzxiong** [matrixorigin/matrixone#12477]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): main b3da04e09
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

UT Test on Ubuntu/x86
- https://github.com/matrixorigin/matrixone/actions/runs/6719843361/job/18262227548?pr=12473

```
2023-11-01T12:55:32.6960759Z ==================
2023-11-01T12:55:32.6961075Z WARNING: DATA RACE
2023-11-01T12:55:32.6961445Z Write at 0x00c00055bd70 by goroutine 928:
2023-11-01T12:55:32.6962196Z   github.com/matrixorigin/matrixone/pkg/lockservice.TestIterLocks.func1.1()
2023-11-01T12:55:32.6963485Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_observability_test.go:182 +0x2a8
2023-11-01T12:55:32.7131821Z
2023-11-01T12:55:32.7132124Z Previous write at 0x00c00055bd70 by goroutine 929:
2023-11-01T12:55:32.7132983Z   github.com/matrixorigin/matrixone/pkg/lockservice.TestIterLocks.func1.2()
2023-11-01T12:55:32.7134270Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_observability_test.go:191 +0x2a8
2023-11-01T12:55:32.7134976Z
2023-11-01T12:55:32.7135143Z Goroutine 928 (running) created at:
2023-11-01T12:55:32.7135846Z   github.com/matrixorigin/matrixone/pkg/lockservice.TestIterLocks.func1()
2023-11-01T12:55:32.7137099Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_observability_test.go:180 +0xe11
2023-11-01T12:55:32.7138184Z   github.com/matrixorigin/matrixone/pkg/lockservice.getRunner.func1.1()
2023-11-01T12:55:32.7139268Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:70 +0x61a
2023-11-01T12:55:32.7140375Z   github.com/matrixorigin/matrixone/pkg/lockservice.runLockServiceTestsWithLevel.func1()
2023-11-01T12:55:32.7141612Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1446 +0x1da
2023-11-01T12:55:32.7142623Z   github.com/matrixorigin/matrixone/pkg/lockservice.RunLockServicesForTest()
2023-11-01T12:55:32.7143896Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/test_helper.go:77 +0xaca
2023-11-01T12:55:32.7144958Z   github.com/matrixorigin/matrixone/pkg/lockservice.runLockServiceTestsWithLevel()
2023-11-01T12:55:32.7146126Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1437 +0x10e
2023-11-01T12:55:32.7147199Z   github.com/matrixorigin/matrixone/pkg/lockservice.runLockServiceTests()
2023-11-01T12:55:32.7148301Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1405 +0x112
2023-11-01T12:55:32.7149259Z   github.com/matrixorigin/matrixone/pkg/lockservice.getRunner.func1()
2023-11-01T12:55:32.7150312Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:47 +0x53
2023-11-01T12:55:32.7151243Z   github.com/matrixorigin/matrixone/pkg/lockservice.TestIterLocks()
2023-11-01T12:55:32.7152423Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_observability_test.go:149 +0xc1
2023-11-01T12:55:32.7153312Z   testing.tRunner()
2023-11-01T12:55:32.7153996Z       /opt/hostedtoolcache/go/1.20.10/x64/src/testing/testing.go:1576 +0x216
2023-11-01T12:55:32.7154705Z   testing.(*T).Run.func1()
2023-11-01T12:55:32.7155406Z       /opt/hostedtoolcache/go/1.20.10/x64/src/testing/testing.go:1629 +0x47
2023-11-01T12:55:32.7155894Z
2023-11-01T12:55:32.7156056Z Goroutine 929 (finished) created at:
2023-11-01T12:55:32.7156772Z   github.com/matrixorigin/matrixone/pkg/lockservice.TestIterLocks.func1()
2023-11-01T12:55:32.7158000Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_observability_test.go:189 +0x1154
2023-11-01T12:55:32.7159079Z   github.com/matrixorigin/matrixone/pkg/lockservice.getRunner.func1.1()
2023-11-01T12:55:32.7160156Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:70 +0x61a
2023-11-01T12:55:32.7161260Z   github.com/matrixorigin/matrixone/pkg/lockservice.runLockServiceTestsWithLevel.func1()
2023-11-01T12:55:32.7162484Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1446 +0x1da
2023-11-01T12:55:32.7163501Z   github.com/matrixorigin/matrixone/pkg/lockservice.RunLockServicesForTest()
2023-11-01T12:55:32.7164594Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/test_helper.go:77 +0xaca
2023-11-01T12:55:32.7266763Z   github.com/matrixorigin/matrixone/pkg/lockservice.runLockServiceTestsWithLevel()
2023-11-01T12:55:32.7268062Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1437 +0x10e
2023-11-01T12:55:32.7269085Z   github.com/matrixorigin/matrixone/pkg/lockservice.runLockServiceTests()
2023-11-01T12:55:32.7270196Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:1405 +0x112
2023-11-01T12:55:32.7271173Z   github.com/matrixorigin/matrixone/pkg/lockservice.getRunner.func1()
2023-11-01T12:55:32.7272226Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_test.go:47 +0x53
2023-11-01T12:55:32.7273169Z   github.com/matrixorigin/matrixone/pkg/lockservice.TestIterLocks()
2023-11-01T12:55:32.7274339Z       /home/runner/work/matrixone/matrixone/pkg/lockservice/service_observability_test.go:149 +0xc1
2023-11-01T12:55:32.7275159Z   testing.tRunner()
2023-11-01T12:55:32.7275829Z       /opt/hostedtoolcache/go/1.20.10/x64/src/testing/testing.go:1576 +0x216
2023-11-01T12:55:32.7276468Z   testing.(*T).Run.func1()
2023-11-01T12:55:32.7277164Z       /opt/hostedtoolcache/go/1.20.10/x64/src/testing/testing.go:1629 +0x47
2023-11-01T12:55:32.7277747Z ==================
...
2023-11-01T12:55:32.7716189Z 2023/11/01 12:54:10.761833 +0000 DEBUG cn-service.lockservice.lock-server application stopped {"uuid": "", "name": "lock-server", "listen-addresses": "[/tmp/639152834.sock]"}
2023-11-01T12:55:32.7717817Z 2023/11/01 12:54:10.761865 +0000 DEBUG cn-service.lockservice.allocator lock service allocator closed {"uuid": ""}
2023-11-01T12:55:32.7718870Z     testing.go:1446: race detected during execution of test
2023-11-01T12:55:32.7719475Z --- FAIL: TestIterLocks (0.12s)
```

**Comment by zhangxu19830126** [comment]
fixed by #13020

**Comment by xzxiong** [comment]
checked.
